crates/abd-clam/src/chaoda/meta_ml.rs

```diff
@@ -0,0 +1,187 @@
+            &[346.999, 193.2, 359.4, 113.270, 1952., 63.639],
+            &[365.385, 187.0, 354.7, 115.094, 1953., 64.989],
+            &[363.112, 357.8, 335.0, 116.219, 1954., 63.761],
+            &[397.469, 290.4, 304.8, 117.388, 1955., 66.019],
+            &[419.180, 282.2, 285.7, 118.734, 1956., 67.857],
+            &[442.769, 293.6, 279.8, 120.445, 1957., 68.169],
+            &[444.546, 468.1, 263.7, 121.950, 1958., 66.513],
+            &[482.704, 381.3, 255.2, 123.366, 1959., 68.655],
+            &[502.601, 393.1, 251.4, 125.368, 1960., 69.564],
+            &[518.173, 480.6, 257.2, 127.852, 1961., 69.331],
+            &[554.894, 400.7, 282.7, 130.081, 1962., 70.551],
+        ]);
+        let dummy_y = vec![
+            83.0, 88.5, 88.2, 89.5, 96.2, 98.1, 99.0, 100.0, 101.2, 104.6, 108.4, 110.8, 112.6, 114.2, 115.7, 116.9,
+        ];
+
+        Ok(match model {
+            "lr" | "LR" | "LinearRegression" => Self::LinearRegression(
+                LinearRegression::fit(
+                    &dummy_x,
+                    &dummy_y,
+                    LinearRegressionParameters::default().with_solver(LinearRegressionSolverName::QR),
+                )
+                .map_err(|e| e.to_string())?,
+            ),
+            "en" | "EN" | "ElasticNet" => Self::ElasticNet(
+                ElasticNet::fit(&dummy_x, &dummy_y, ElasticNetParameters::default()).map_err(|e| e.to_string())?,
+            ),
+            "la" | "LA" | "Lasso" => {
+                Self::Lasso(Lasso::fit(&dummy_x, &dummy_y, LassoParameters::default()).map_err(|e| e.to_string())?)
+            }
+            "rr" | "RR" | "RidgeRegression" => Self::RidgeRegression(
+                RidgeRegression::fit(&dummy_x, &dummy_y, RidgeRegressionParameters::default())
+                    .map_err(|e| e.to_string())?,
+            ),
+            "dt" | "DT" | "DecisionTreeRegressor" => Self::DecisionTreeRegressor(
+                DecisionTreeRegressor::fit(&dummy_x, &dummy_y, DecisionTreeRegressorParameters::default())
+                    .map_err(|e| e.to_string())?,
+            ),
+            "rf" | "RF" | "RandomForestRegressor" => Self::RandomForestRegressor(
+                RandomForestRegressor::fit(&dummy_x, &dummy_y, RandomForestRegressorParameters::default())
+                    .map_err(|e| e.to_string())?,
+            ),
+            _ => return Err(format!("Unknown model: {model}")),
+        })
+    }
+
+    /// Get the default models.
+    #[must_use]
+    pub fn defaults() -> Vec<Self> {
+        vec![
+            Self::new("LR").unwrap_or_else(|e| unreachable!("{e}")),
+            // Self::new("EN").unwrap_or_else(|e| unreachable!("{e}")),
+            // Self::new("LA").unwrap_or_else(|e| unreachable!("{e}")),
+            // Self::new("RR").unwrap_or_else(|e| unreachable!("{e}")),
+            Self::new("DT").unwrap_or_else(|e| unreachable!("{e}")),
+            // Self::new("RF").unwrap_or_else(|e| unreachable!("{e}")),
```

Comment:
I will bring out the rest of these after some experimentation with the training routine

Reply:
Ok, no need to hold up the PR for this but let's not forget.